cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/sqlite/SQLDatabaseFactory.java

```diff
@@ -70,106 +72,69 @@ private static boolean isFtsAvailable() {
     }
 
     /**
-     * SQLCipher-based implementation for creating database.
-     * @param dbFilename full file path of the db file
+     * <p>
+     * Open or create a database file, and return a connection to it, optionally backed by a
+     * SQLCipher enabled database.
+     * </p>
+     * <p>
+     * If the database file does not exist, it will be created, and any intermediate directories
+     * will be created as necessary.
+     * </p>
+     * @param dbFile full file path of the db file
      * @param provider Key provider object storing the SQLCipher key
      *                 Supply a NullKeyProvider to use a non-encrypted database.
      * @return {@code SQLDatabase} for the given filename
-     * @throws IOException if the file does not exists, and also
-     *         can not be created
+     * @throws IOException if the file cannot be opened, or the database files or its intermediate
+     *                     directories cannot be created.
      * @throws SQLException if the database cannot be opened.
      */
-    public static SQLDatabase createSQLDatabase(String dbFilename, KeyProvider provider) throws IOException, SQLException {
-        makeSureFileExists(dbFilename);
-        return internalCreateSQLDatabase(dbFilename, provider);
+    public static SQLDatabase openOrCreateSQLDatabase(File dbFile, KeyProvider provider) throws IOException, SQLException {
```

Comment:
Again, this really sounds like its only going to open the database if it already exists, `openSQLDatabase` feels alot better to my mind.